Assets/HoloToolkit/Input/Scripts/KeywordManager.cs

```diff
@@ -84,26 +102,30 @@ void OnDestroy()
             }
         }
 
+        /// <summary>
+        /// Listens for key presses and calls the corrisponding keyword event
+        /// </summary>
         private void ProcessKeyBindings()
         {
-            foreach (var kvp in KeywordsAndResponses)
+            for (int i = 0; i < KeywordsAndResponses.Length; i++)
             {
-                if (Input.GetKeyDown(kvp.KeyCode))
+                if (Input.GetKeyDown(KeywordsAndResponses[i].KeyCode) && OnKeywordRecognized != null)
                 {
-                    kvp.Response.Invoke();
+                    OnKeywordRecognized(KeywordsAndResponses[i].Keyword);
                     return;
                 }
             }
         }
 
+        /// <summary>
+        /// Throws the OnKeywordRecognized event if the keyword is registered in the responses list.
+        /// </summary>
+        /// <param name="args">The recognized keyword.</param>
         private void KeywordRecognizer_OnPhraseRecognized(PhraseRecognizedEventArgs args)
         {
-            UnityEvent keywordResponse;
-
-            // Check to make sure the recognized keyword exists in the methods dictionary, then invoke the corresponding method.
-            if (responses.TryGetValue(args.text, out keywordResponse))
+            if (OnKeywordRecognized != null && responses.Contains(args.text))
             {
-                keywordResponse.Invoke();
+                OnKeywordRecognized(args.text);
```

Comment:
It also causes me to need a specific method for handling keywords in various scripts ([like this one you mentioned above](https://github.com/HodgsonSDAS/HoloToolkit-Unity/blob/3a4549ab0092827f7539b0b77e0da72513ffc4e8/Assets/HoloToolkit/Input/Scripts/Interactable.cs#L81)). In the current implementation, I can call any public method I want as the event, allowing me to reuse code easily instead of routing it through another method.